test/csuite/incr_backup/main.c

```diff
@@ -0,0 +1,846 @@
+ */
+static void
+active_files_add(ACTIVE_FILES *active, const char *name)
+{
+    uint32_t pos;
+
+    pos = active->count++;
+    active->names = drealloc(active->names, sizeof(char *) * active->count);
+    active->names[pos] = strdup(name);
+}
+
+/*
+ * active_files_sort_function --
+ *     Sort function for qsort.
+ */
+static int
+active_files_sort_function(const void *left, const void *right)
+{
+    return (strcmp(*(const char **)left, *(const char **)right));
+}
+
+/*
+ * active_files_sort --
+ *     Sort the list of names in the active file list.
+ */
+static void
+active_files_sort(ACTIVE_FILES *active)
+{
+    __wt_qsort(active->names, active->count, sizeof(char *), active_files_sort_function);
+}
+
+/*
+ * active_files_remove_missing --
+ *     Files in the previous list that are missing from the current list are removed.
+ */
+static void
+active_files_remove_missing(ACTIVE_FILES *prev, ACTIVE_FILES *cur, const char *dirname)
+{
+    uint32_t curpos, prevpos;
+    int cmp;
+    char filename[1024];
+
+    active_files_print(prev, "computing removals: previous list of active files");
+    active_files_print(cur, "computing removals: current list of active files");
+    curpos = 0;
+    /*
+     * Walk through the two lists looking for non-matches.
+     */
+    for (prevpos = 0; prevpos < prev->count; prevpos++) {
+again:
+        if (curpos >= cur->count)
+            cmp = -1; /* There are extra entries at the end of the prev list */
+        else
+            cmp = strcmp(prev->names[prevpos], cur->names[curpos]);
+
+        if (cmp == 0)
+            curpos++;
+        else if (cmp < 0) {
+            /*
+             * There is something in the prev list not in the current list. Remove it, and continue
+             * - don't advance the current list.
+             */
+            testutil_check(
+              __wt_snprintf(filename, sizeof(filename), "%s/%s", dirname, prev->names[prevpos]));
+            VERBOSE(3, "Removing file from backup: %s\n", filename);
+            remove(filename);
+        } else {
+            /*
+             * There is something in the current list not in the prev list. Walk past it in the
+             * current list and try again.
+             */
+            curpos++;
+            goto again;
+        }
+    }
+}
+
+/*
+ * active_files_free --
+ *     Free the list of active files.
+ */
+static void
+active_files_free(ACTIVE_FILES *active)
+{
+    uint32_t i;
+
+    for (i = 0; i < active->count; i++)
+        free(active->names[i]);
+    free(active->names);
+    active_files_init(active);
+}
+
+/*
+ * active_files_move --
+ *     Move an active file list to the destination list.
+ */
+static void
+active_files_move(ACTIVE_FILES *dest, ACTIVE_FILES *src)
+{
+    active_files_free(dest);
+    *dest = *src;
+    WT_CLEAR(*src);
+}
+
+/*
+ * table_updates --
+ *     Potentially make changes to a single table.
+ */
+static void
+table_updates(WT_SESSION *session, TABLE *table)
```

Comment:
Can you consider a slightly different name for the function? I thought it referred specifically to `cursor->update`. Maybe `table_mods` or `table_changes` or `do_work` ...?